Source/Details/ASCollectionGalleryLayoutDelegate.mm

```diff
@@ -105,6 +110,31 @@ + (ASCollectionLayoutState *)calculateLayoutWithContext:(ASCollectionLayoutConte
   NSMutableArray<_ASGalleryLayoutItem *> *children = ASArrayByFlatMapping(elements.itemElements,
                                                                           ASCollectionElement *element,
                                                                           [[_ASGalleryLayoutItem alloc] initWithItemSize:itemSize collectionElement:element]);
+  NSInteger numberOfSupplementaryelementsBefore = 0;;
+  if ([context.elements.supplementaryElementKinds count]) {
```

Comment:
`context.elements` is stored in an `elements` variable at the beginning of this method. For readability, please use it throughout this method instead of reaching out to the context every time.